MatrixSDKTests/MXSessionTests.m

```diff
@@ -65,6 +66,71 @@ - (void)tearDown
     [super tearDown];
 }
 
+// Check MXSession clears initial sync cache after handling sync response.
+//
+// - Have Bob start a new session
+// - Run initial sync on Bob's session
+// -> The initial sync cache must be used
+- (void)testInitialSyncSuccess
+{
+    [matrixSDKTestsData doMXSessionTestWithBob:self readyToTest:^(MXSession *mxSession, XCTestExpectation *expectation) {
+
+        id<MXSyncResponseStore> cache = [[MXSyncResponseFileStore alloc] initWithCredentials:matrixSDKTestsData.bobCredentials];
+        XCTAssertEqual(cache.outdatedSyncResponseIds.count, 0, @"Initial sync cache must be reset after successful initialization");
+
+        [expectation fulfill];
+    }];
+}
+
+// Check MXSession updates initial sync cache when an error occurs handling the sync response.
+//
+// - Have Bob start a new session
+// - Run initial sync on Bob's session
+// - In the middle of the process, pause the session
```

Comment:
Closing the session and deleting the MXFileStore would be closer to the problem we are trying to improve.

Reply:
Updated to close the session, no need to delete data in store, it's already using a `MXNoStore` instance.